### ISSUE TITLE
Unable to complete authentication when using SFSafariViewController webview type

### ISSUE THREAD

**Issue by tobihagemann** [AzureAD/microsoft-authentication-library-for-objc#724]
I noticed that my authentication routine doesn't work when running it on iOS 10. When trying to acquire a token, the interactive controller shows up, I enter my credentials, and submit them. But then it gets stuck and stays open. All I can do is tap "Done" at the top left to dismiss the screen (but that leads to a "cancel").

I can reproduce the issue on iOS 13 when using `MSALWebviewTypeSafariViewController`. That's why I'm guessing that this webview type is maybe buggy? Or is there anything that I have to consider so that the interactive controller gets finished and dismissed?

Small example:

```objc
MSALWebviewParameters *webviewParameters = [[MSALWebviewParameters alloc] initWithParentViewController:viewController];
webviewParameters.webviewType = MSALWebviewTypeSafariViewController; // enforce SFSafariViewController
MSALInteractiveTokenParameters *parameters = [[MSALInteractiveTokenParameters alloc] initWithScopes:@[] webviewParameters:webviewParameters];
[self.publicClientApplication acquireTokenWithParameters:parameters completionBlock:^(MSALResult *result, NSError *error) {
  // do something
}];
```

**Comment by oldalton** [comment]
@tobihagemann, please make sure you have applied these 2 steps:
1. Call MSAL from your AppDelegate: https://github.com/AzureAD/microsoft-authentication-library-for-objc#swift-1
2. Added your redirect scheme to Info.plist: https://github.com/AzureAD/microsoft-authentication-library-for-objc#ios-only-steps

I've just tested SafariViewController on iOS 13 and it seems working. SafariViewController response handling relies on URL schemes, so if you are missing any of these 2 steps, your app would never receive response. Please double check the steps and let me know if both of them are followed. 

If after following the steps you're still seeing the issue, please collect a verbose MSAL log for the next steps: https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-logging#logging-in-msal-for-ios-and-macos. Thanks!

**Comment by tobihagemann** [comment]
Thank you for pointing me in the right direction. I found my issue, I was checking for the "wrong" scheme inside my app delegate.

I followed the instructions and used this pattern for my redirect URI scheme: `msauth.[BUNDLE_ID]`. However, my bundle id contains uppercase letters and they got lowercased inside the app delegate even though they're defined with uppercase in Azure portal. I guess, URI schemes are not case-preserving. :wink: That's why my `isEqualToString` check failed.

It works now, thank you!
